src/stdlib_hashmap_wrappers.f90

```diff
@@ -0,0 +1,378 @@
+
+        module procedure set_char_key,   &
+                         set_int8_key,   &
+                         set_other
+
+    end interface set
+
+contains
+
+
+    pure subroutine copy_key( key_in, key_out )
+!! Version: Experimental
+!!
+!! Copies the contents of the key, key_in, to the key, key_out
+!! Arguments:
+!!     key_in  - the input key
+!!     key_out - the output key
+        type(key_type), intent(in)  :: key_in
+        type(key_type), intent(out) :: key_out
+
+        key_out % value = key_in % value
+
+    end subroutine copy_key
+
+
+    subroutine copy_other( other_in, other_out )
+!! Version: Experimental
+!!
+!! Copies the other data, other_in, to the variable, other_out
+!! Arguments:
+!!     other_in  - the input data
+!!     other_out - the output data
+        type(other_type), intent(in)  :: other_in
+        type(other_type), intent(out) :: other_out
+
+        allocate(other_out % value, source = other_in % value )
+
+    end subroutine copy_other
+
+
+    function equal_keys( key1, key2 ) result(test) ! Chase's tester
+!! Version: Experimental
+!!
+!! Compares two keys for equality
+!! Arguments:
+!!     key1 - the first key
+!!     key2 - the second key
+        logical                    :: test
+        type(key_type), intent(in) :: key1
+        type(key_type), intent(in) :: key2
+
+        if ( size(key1 % value, kind=int64) /= &
+             size(key2 % value, kind=int64) ) then
+            test = .false.
+            return
+        end if
+
+        if ( all( key1 % value == key2 % value ) ) then
+            test = .true.
+        else
+            test = .false.
+        end if
+
+    end function equal_keys
+
+
+    subroutine free_key( key )
+!! Version: Experimental
+!!
+!! Frees the memory in a key
+!! Arguments:
+!!     key  - the key
+        type(key_type), intent(inout) :: key
+
+        if ( allocated( key % value ) ) deallocate( key % value )
+
+    end subroutine free_key
+
+
+    subroutine free_other( other )
+!! Version: Experimental
+!!
+!! Frees the memory in the other data
+!! Arguments:
+!!     other  - the other data
+        type(other_type), intent(inout) :: other
+
+        if ( allocated( other % value) ) deallocate( other % value )
+
+    end subroutine free_other
+
+
+    subroutine get_char_key( key, value )
+!! Version: Experimental
+!!
+!! Gets the contents of the key as a CHARACTER string
+!! Arguments:
+!!     key   - the input key
+!!     value - the contents of key mapped to a CHARACTER string
+        type(key_type), intent(in)             :: key
+        character(:), allocatable, intent(out) :: value
+
+        integer(int64) :: key_as_char
+        integer(int64) :: key_size
+
+        key_size = size( key % value, kind=int64 )
+        select case( bytes_char )
+        case(1)
+            key_as_char = key_size
+        case(2)
+            if ( iand( key_size, 1_int64 ) > 0 ) then
+                stop 'KEY does not map to a character string.'
```

Comment:
[Same as #r865894471](https://github.com/fortran-lang/stdlib/pull/611/files#r865894471) and the other cases around here

Reply:
Ok, such stops can only occur if the user environment/compiler/combination-of-flags make the `character_storage_size` unusual. I guess you should place some comments here to make sure the reader is aware that this checking is happening.
I see now how the Unicode thing started.

Reply:
This is mapping a byte array to a string. `case(2)` is when the number of bytes in a default character is two. (I believe that can occur in MS Windows.) In that case, the `key_size` can never be an odd number. Either there is a problem with the logic of the rest of the code, or a bit error has occurred. This is unrecoverable. This might be clearer by making the test `and( key_size, 1_int64 ) == 1`.

Reply:
Then it would make sense to change the `stop` message to something less cryptic and leave some comments about how such an edge case could ever happen. This is probably like an ICE - internal compiler error, an inconsistency that is completely unrecoverable.

I can think something like:
```suggestion
                error stop 'Internal Error at stdlib_hashmaps: System uses 2 bytes per character, so key_size can't be an odd number'
```

Reply:
If there is a consistency check it means it can be triggered somehow (incorrect code when developing for example), otherwise, it shouldn't be part of the code in the first place and compilers just optimize it off: Ex: `if (.false.) stop`.
If such a thing ever happens it would be good for anyone experiencing that to know that is something related to an inconsistent internal state and not something else (like incorrect inputs).
Does this make any sense?